### PR TITLE
Fixing single candidate_label return.

### LINKED ISSUE #24008: Zero-shot image classification with single-label results in 'float is not iterable' error
### System Info

- `transformers` version: 4.29.2
- Platform: Linux-5.15.107+-x86_64-with-glibc2.31
- Python version: 3.10.11
- Huggingface_hub version: 0.15.1
- Safetensors version: not installed
- PyTorch version (GPU?): 2.0.1+cu118 (False)
- Tensorflow version (GPU?): 2.12.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.6.9 (cpu)
- Jax version: 0.4.10
- JaxLib version: 0.4.10
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@Narsil

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

**Colab notebook**: https://colab.research.google.com/gist/josephrocca/fdf9537e06fd15d2b5a81727fd70d56b/zero-shot-classification-with-single-label.ipynb

```py
!pip install transformers
!wget https://i.imgur.com/RKsLoNB.png

from transformers import pipeline
image_classifier = pipeline("zero-shot-image-classification", model="openai/clip-vit-large-patch14-336")
text_classifier = pipeline(model="facebook/bart-large-mnli")

# multi-label text - works ✅
text_classifier("houston, we have a problem with the thruster", candidate_labels=["astronaut", "forest cabin", "rabbit and lion"])

# single-label text - works ✅
text_classifier("houston, we have a problem with the thruster", candidate_labels=["astronaut"])

# multi-label image - works ✅
image_classifier("RKsLoNB.png", candidate_labels = ["astronaut", "forest cabin", "rabbit and lion"])

# single-label image - doesn't work ❌
image_classifier("RKsLoNB.png", candidate_labels = ["astronaut"])
```

### Expected behavior

Like in the text case, I'd expect it to not give an error, and importantly, I'd expect it to give a value between 0 and 1, rather than giving a value of exactly 1 (again, like in the text classification case). In other words, if you provide only 1 label, the scores change from being *relative to other label scores* to being a simple **similarity score** between the text and the image - i.e. a binary classification (with the underlying score exposed so you can choose your own threshold).

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fix #24008

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks!

**Review by NielsRogge — APPROVED** [review]
